File: Solution/Medium/2487 Remove Nodes From Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNodes(ListNode head) {
        Stack<ListNode> stack = new Stack<>();
        while (head != null) {
            stack.push(head);
            head = head.next;
        }
        ListNode res = stack.pop();
        while(!stack.isEmpty()) {
            ListNode curr = stack.pop();
            if (res.val > curr.val) {
                continue;
            }
            curr.next = res;
            res = curr;
        }
        return res;
    }
}

File: Solution/Medium/2487 Remove Nodes From Linked List.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun removeNodes(head: ListNode?): ListNode? {
        val stack = ArrayDeque<ListNode>()
        var temp = head;
        while (temp != null) {
            stack.addLast(temp)
            temp = temp.next
        }
        var res = stack.removeLast()
        while (!stack.isEmpty()) {
            val curr = stack.removeLast()
            if (res.`val` <= curr.`val`) {
                curr.next = res;
                res = curr;
            }
        }
        return res;
    }
}
